cache/memcache/memcache.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package memcache provides an implementation of prototransform.Cache
+// that is backed by a memcached instance: https://memcached.org/.
+package memcache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/bufbuild/prototransform"
+)
+
+type Config struct {
+	Client            *memcache.Client
+	KeyPrefix         string
+	ExpirationSeconds int32
```

Comment:
Why not use `time.Duration` here as well just like the redis one?

Reply:
Because memcache only supports expiry in seconds. It seemed less offensive to round to milliseconds for redis. But providing a `time.Duration`, in my opinion, implies more than seconds precision. For milliseconds, it seemed okay 🤷. If you think it's too inconsistent, I'd rather change redis to `ExpirationMillis` than change this to `time.Duration`.